### DISCUSSION TITLE
despite using enabled it stills fetched the data

### DISCUSSION THREAD

**Discussion by akmittal8171** [TanStack/query#2430] [General]
I am loading the data on condition based after searching a lot, I didn't find replacement of enabled as its not working.According to the docs, it should work. Any ideas what to do

const {
        isLoading: loader,
        data,
    } = useQuery('posts', () => getPosts("87"), {
        staleTime: 1000 * 60 * 10, 
        cacheTime: 1000 * 60 * 15, 
        keepPreviousData: true,
    },{
        enabled:false
    })

**Comment by TkDodo** [comment]
I can assure you that enabled works as advertised. Can you put your code in a codesandbox to show us what isn’t working please?
